src/osx/cocoa/nonownedwnd.mm

```diff
@@ -612,7 +613,43 @@ - (void)windowWillEnterFullScreen:(NSNotification *)notification
         [view setFrameSize: expectedframerect.size];
     }
 }
- 
+
+// from https://developer.apple.com/library/archive/documentation/GraphicsAnimation/Conceptual/HighResolutionOSX/CapturingScreenContents/CapturingScreenContents.html
+
+- (void)windowDidChangeBackingProperties:(NSNotification *)notification
+{
+    NSWindow* theWindow = (NSWindow*)[notification object];
+    wxNonOwnedWindowCocoaImpl* windowimpl = [theWindow WX_implementation];
+    wxNonOwnedWindow* wxpeer = windowimpl ? windowimpl->GetWXPeer() : NULL;
+    if (wxpeer)
+    {
+        CGFloat newBackingScaleFactor = [theWindow backingScaleFactor];
+        CGFloat oldBackingScaleFactor = [[[notification userInfo]
+                                          objectForKey:@"NSBackingPropertyOldScaleFactorKey"]
+                                         doubleValue];
+        if (newBackingScaleFactor != oldBackingScaleFactor)
+        {
+            wxSize oldDPI((int)(oldBackingScaleFactor*72.0),(int)(oldBackingScaleFactor*72.0));
+            wxSize newDPI((int)(newBackingScaleFactor*72.0),(int)(newBackingScaleFactor*72.0));
```

Comment:
Should I change those to use `wxRound()` too? It seems it would be better, if only for consistency with `wxWindow::GetDPI()`?

Reply:
OK, I've done this and force pushed my changes here, I hope it's ok.

Reply:
yes, perfect, thanks